pkg/auth/middleware_test.go

```diff
@@ -0,0 +1,126 @@
+package auth_test
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/guidewire/fern-reporter/pkg/auth"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+	"net/http"
+	"net/http/httptest"
+)
+
+var _ = Describe("JWK Management", func() {
+	var (
+		server *ghttp.Server
+		url    string
+	)
+
+	BeforeEach(func() {
+		// Create a fake server to simulate JWK provider
+		server = ghttp.NewServer()
+		url = server.URL()
+	})
+
+	AfterEach(func() {
+		server.Close()
+	})
+
+	Describe("UpdateJWKS", func() {
+		Context("when fetching JWKs is successful", func() {
+			BeforeEach(func() {
+				server.AppendHandlers(
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/"),
+						ghttp.RespondWithJSONEncoded(http.StatusOK, map[string]interface{}{
+							"keys": []interface{}{
+								map[string]interface{}{
+									"kty": "RSA",
+									"alg": "RS256",
+									"kid": "mAMF03ZNwGBz54bjNJLGtlTC9oP8zJSLrpkfBIH1R-E",
+									"use": "sig",
+									"e":   "AQAB",
+									"n":   "2uCExuw6kt86vt28clwQ8d0C1UHMUFUbBlthwiOpTTQYkFSbBUQKBJ16P9pnBrVwVr6-s1-84SKGnJnK6EX6IuiTKJQeEurV67ivoahtZXFBk02fBWd8LrkmDdCE59EsVB8zmHycYMCjm133n1THXjcpjQXKHWmTr3D7mP0jgGZWSdxTgGuWbglX5_OhqEZy7LNQQQYwBnGTsBxCm9Fr6g9b_dWz7l_pXpuVuaesMhL7zahwwCBE6d-tpcN_jhujTT6UhRB63uQsehchAot1BWNdBRsOtQtt4OW9EGqUD8ebVtAt8wchRi6wjCva9MLXQQNWehQftSTRqHZ8HNIOsw",
+								},
+							},
+						}),
+					),
+				)
```

Comment:
It would be cleaner to use the JWK struct here, instead of `map[string]interface{}` inline.

Reply:
This was changed in the refactor.